### PR TITLE
Add the possiblity to add more or all locales (solves #122)

### PR DESCRIPTION
**Pull request by soriyath**
**Description of the change**

This PR adds the possibility to add extra locales to the default ones or add all the supported locales.

It defaults the previous behaviour (i.e. inclusion of `en_AU.UTF-8 UTF8` and `en_US.UTF-8 UTF8`.
Additionally, one can add more locales using the new build arg `EXTRA_LOCALES` using entries from `/usr/share/i18n/SUPPORTED` and separating several entries with a comma and a space, like so: `"fr_CH.UTF-8 UTF-8, de_CH.UTF-8 UTF-8, it_CH.UTF-8 UTF-8, de_DE.UTF-8 UTF-8"` or add all the locales using the new build arg `ALL_LOCALES` set to `1`.

**Benefits**

This PR makes it easier for non English speaker to use the Moodle container and further broadens its audience, without putting existing users at risk by defaulting the same behaviour it had before.

**Possible drawbacks**

Adding all the locales makes the build time slower. This is a one time cost since one doesn't build Docker images repetitively, notwithstanding Dockerfile changes.

**Applicable issues**

Any non English speaker or any system administrator that provide services to multi-language audience will be thrilled with this PR.

**Additional information**

This PR relates to this issue on the Bitnami Helm Chart: https://github.com/bitnami/charts/issues/2493

### CODE REVIEW

**Comment by rafariossaa** [comment]
Hi,
  Sorry for the delay.
  We are going to integrate these change into our CI/CD so it can be automatically included in following releases.
  We will come back to you.

**Comment by soriyath** [comment]
@rafariossaa i'm not sure i understand what you mean? are you considering merging my work?

**Comment by rafariossaa** [comment]
@soriyath ,
  Yes, we are considering it.

**Comment by soriyath** [comment]
@rafariossaa @andresbono what's the status on this?

I believe it makes sense to merge because the defaults fall back on the previous behaviour and it adds the possibility to add new locales if needed.

it can help a lot of people who speak other languages than english, imho.

**Comment by rafariossaa** [comment]
Hi,
  Sorry for the delay, I can see you point. The task is in our backlog. We hope to come back soon, but not ETA yet.

**Comment by marcosbc** [comment]
Hi, just a note that we recently refactored the initialization logic of Moodle, and it is now based on Bash scripts in the `rootfs` folder in the root of the repository. For more details see: https://github.com/bitnami/bitnami-docker-moodle#390-debian-10-r17

I have a question related to this PR, isn't it easier to extend the image with the custom locales? I.e.:

```
FROM bitnami/moodle:3.9.0-debian-10-r17

RUN echo 'es_ES.UTF-8 UTF-8' >> /etc/locale.gen
```

And that way no change is needed to the Docker image itself, since you would just be [customizing the image for your own use](https://github.com/bitnami/bitnami-docker-moodle#customize-this-image). Documenting how to install specific locales is great though, based on the customization guide.

**Comment by soriyath** [comment]
I can adapt the change to the new approach, no worries.

@marcosbc I think you are wrong. I understand it would make it easier from your point of view. "_Customizing the image for your own use_" really means that you are delegating a basic feature to non english speaker (basically: let them handle stuff, we won't do anything about it).

Non english speaker are not second class citizens.

By defaulting to the existing behaviour, there is literally no changes for existing users and english users that don't need internationalization.
Adding this capability means one can build the Docker containers with the extra needed locales, or all of them (which, IMHO, would be even better than the currents defaults! makes more sense to have all locales than only English).

In the Helm hart, we could add a feature flag that would toggle building the image on the fly vs downloading a ready made container (we could also have a ready made container with all the locales for that matter).

The bitnami docker containers and helm chart makes it easy for many people to just use the software.
Making the experience richer upstream benefits everybody.
If it were not the case, why bother at all? Everyone could just maintain their own Dockerfiles or build locally on physical or virtual machines...

**Comment by marcosbc** [comment]
Feel free to update the PR and we'll continue discussions internally on whether the changes make sense to be implemented.

I think adding a Docker ARG should not be problematic, so if that allows users to use the image with more ease, then be it. However, in the end, users will still need to build the image for themselves, so I really don't see how asking them to create their own Dockerfile based on the other one is more complex/problematic.

> Adding this capability means one can build the Docker containers with the extra needed locales, or all of them (which, IMHO, would be even better than the currents defaults! makes more sense to have all locales than only English).

The reason for not adding all locales is to save disk space. Adding all locales could easily add 200M+ to the container image, so in that case I'd prefer if users add the locales they want.

>In the Helm hart, we could add a feature flag that would toggle building the image on the fly vs downloading a ready made container (we could also have a ready made container with all the locales for that matter).

That I'm not so convinced about, it would be very specific and we're trying to tie all Helm chart releases to a specific Bitnami container image revision, to ensure the image works with the specific template.

**Comment by soriyath** [comment]
@marcosbc forget about it.

I think adding support for more languages for a user base that spans the entire planet is a no brainer. If we disagree on that there's not much to be discussed, IMHO.

I'll fork the project and maintain my own charts.

**Comment by marcosbc** [comment]
@soriyath Sorry for my previous message. There has been a misunderstanding and most likely due to how I expressed myself.

My main concerns were related to the tradeoffs of adding more languages by default, and the consequent size increase of the container (+200M). Also, adding new languages was already possible by extending the Dockerfile with a few lines. However, we understand this means leaving the burden of maintaining a Dockerfile to the user, which can be greatly simplified by your PR by allowing them to specify their desired languages at build-time via Docker ARGs.

We're going to re-open this PR and update it with the new Bash approach on our side. This specific change is perfectly fine for us, as it allows users to add more functionality to their Moodle installations, and we'll work on including this change during this week.

File: 3/debian-10/rootfs/opt/bitnami/scripts/locales/add-extra-locales.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
```
Comment (marcosbc) [inline 3/debian-10/rootfs/opt/bitnami/scripts/locales/add-extra-locales.sh:1]: This script is shared with containers that don't contain the Bash libraries (i.e. `liblog.sh`) so unfortunately we cannot use it here.

**Pull request merged**